File: src/lib/Connection.java
package lib;

public class Connection {
  private Node first;
  private Node second;
  private int weight;

  public Connection(Node first, Node second, int weight) {
    this.first = first;
    this.second = second;
    this.weight = weight;
  }

  public Node getFirst() {
    return this.first;
  }

  public Node getSecond() {
    return this.second;
  }

  public int getWeight() {
    return this.weight;
  }

  public void print() {
    System.out.println(this.first.getNode() + " -> " + this.second.getNode() + " = " + this.weight);
  }
}

File: src/backend/lib/connection.py
class Connection:
  def __init__(self, first, second, cost):
    self.first = first
    self.second = second
    self.cost = cost
  
  def __lt__(self, other):
    return self.cost < other.cost
  
  def print(self):
    print(f"{self.first.node} -> {self.second.node} = {self.cost}")